moveit_planners/pilz_industrial_motion_planner/src/trajectory_generator.cpp

```diff
@@ -214,23 +214,25 @@ void TrajectoryGenerator::validateRequest(const planning_interface::MotionPlanRe
   checkGoalConstraints(req.goal_constraints, req.start_state.joint_state.name, req.group_name);
 }
 
-void TrajectoryGenerator::convertToRobotTrajectory(const trajectory_msgs::JointTrajectory& joint_trajectory,
+void TrajectoryGenerator::convertToRobotTrajectory(const planning_scene::PlanningSceneConstPtr& scene, 
+                                                   const trajectory_msgs::JointTrajectory& joint_trajectory,
                                                    const moveit_msgs::RobotState& start_state,
                                                    robot_trajectory::RobotTrajectory& robot_trajectory) const
 {
-  moveit::core::RobotState start_rs(robot_model_);
+  moveit::core::RobotState start_rs(scene->getCurrentState());
```

Comment:
What't the purpose of this change? `robotStateMsgToRobotState` overwrites start_rs two lines below, so this line only default-inializes a RobotState?

Reply:
Now that you mention it. You are right, actually, it does not make much sense, even with the default-initialized `RobotState` using the `robot_model_` should work but the problem was the third parameter of the `robotStateMsgToRobotState`, it is not copying the attached collision objects.

Reply:
> robotStateMsgToRobotState overwrites start_rs two lines below, so this line only default-inializes a RobotState?

I disagree! If the RobotState message does not specify all joints (it is allowed to do that, esp. when `is_diff == true`), all other joint values should be taken from the passed scene and not be set to default! This was another undiscovered issue in this code. Please add this change back to your PR :-)